### PR TITLE
Feature/create Intake commands

### LINKED ISSUE #28: Write basic commands for Intake
SpinIntakeCommand
- Should never end on its own.
- Should take a double (not a supplier). 
- Should `require` its subsystem.

StopSpinIntakeCommand
- Should never end on its own.
- Should `require` its subsystem.

ExtendIntakeCommand & RetractIntakeCommand & ToggleExtensionIntakeCommand
- Should extend `InstantCommand`
- Should `require` its subsystem.

### PR DESCRIPTION
**Pull request by kchenturtles**
Wrote Intake commands SpinIntakeCommand, StopSpinIntakeCommand, ExtendIntakeCommand, RetractIntakeCommand, ToggleExtensionIntakeCommand

Closes #28

### CODE REVIEW

**Review by aidan-mundy — APPROVED** [review]

File: src/main/java/frc/robot/commands/intake/ExtendIntakeCommand.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  /**
+   * Creates an instantly run command that extends the given {@link Intake}.
+   * 
+   * @param intake The {@link Intake} to control.
+   */
+  public ExtendIntakeCommand(Intake intake) {
+    super(() -> {
+      intake.setExtended();
+    }, intake);
```
Comment (citrus-melon) [inline src/main/java/frc/robot/commands/intake/ExtendIntakeCommand.java]: Would it be better to use the double colon operator here (and for the other instant commands)? This isn't a requested change, I'm not sure about this, should probably ask @aidan-mundy 

Like this:
```suggestion
    super(intake::setExtended, intake);
```

File: src/main/java/frc/robot/commands/intake/SpinIntakeCommand.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  public final Intake intake;
+  public final double speed;
+
+  /**
+   * Creates a command that spins the {@link Intake} at a given speed.
+   *
+   * @param intake The {@link Intake} to control
+   * @param speed The speed at which to spin the intake
+   */
+  public SpinIntakeCommand(Intake intake, double speed) {
```
Comment (citrus-melon) [inline src/main/java/frc/robot/commands/intake/SpinIntakeCommand.java:24]: Should this command also have a constructor that takes a supplier for the speed?

**Review by citrus-melon — COMMENTED** [review]
Everything seems good, but I had a few things I wasn't sure about, so I'm not approving yet.

File: src/main/java/frc/robot/commands/intake/SpinIntakeCommand.java (hunk 2 above)
Comment (kchenturtles) [reply src/main/java/frc/robot/commands/intake/SpinIntakeCommand.java:24]: the ticket said "takes a double (not a supplier)" but idk maybe?

Comment (citrus-melon) [reply src/main/java/frc/robot/commands/intake/SpinIntakeCommand.java:24]: Oh. Missed that, sorry!

File: src/main/java/frc/robot/commands/intake/ExtendIntakeCommand.java (hunk 1 above)
Comment (aidan-mundy) [reply src/main/java/frc/robot/commands/intake/ExtendIntakeCommand.java]: Huh, didnt know that was a feature in java..... Yup, definitely better. Thanks @citrus-melon. Please implement this @kchenturtles

**Review by aidan-mundy — CHANGES_REQUESTED** [review]
Suggestion from @citrus-melon needs to be implemented

**Review by citrus-melon — APPROVED** [review]
Looks good!

**Review by aidan-mundy — APPROVED** [review]

**Pull request merged**